src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    let temp = null;
+
+    if (i === 0) {
+      temp = { ...state };
+    } else {
+      temp = { ...res[i - 1] };
+    }
+
+    switch (actions[i].type) {
+      case 'clear':
+        temp = {};
+        break;
+
+      case 'addProperties':
+        for (const addKey in actions[i].extraData) {
+          temp[addKey] = actions[i].extraData[addKey];
+        }
+        break;
+
+      case 'removeProperties':
+        if (Object.keys(temp).length === 1) {
+          temp = {};
+        } else {
```

Comment:
what happens if we pass the wrong property name to delete it will delete the correct one
do we really want this logic? (i don't think so)